File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    last_painted = 0  # 마지막으로 칠해진 지점
    
    for s in section:
        # 현재 구역이 마지막으로 칠한 범위 밖에 있다면
        if s > last_painted:
            answer += 1 
            last_painted = s + m - 1  # 현재 구역부터 m만큼 칠함
            
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int last_painted = 0; // 칠해진 구역의 번호 (1 ~ n)
        
        for(int s : section){
            if(last_painted < s){
                last_painted = s + m - 1; // 룰러로 페인트를 1번 칠함
                answer++;
            }
        }
        
        return answer;
    }
}